cosign/README.md

```diff
@@ -0,0 +1,81 @@
+`sigstore-go` is built as a modular Go codebase including cryptographic signature creation and validation, integration with external services (Fulcio for certificates and Rekor for record-keeping) and API layers.
+The `ratify-verifier-go` implementation shares the same underlying library, `sigstore-go`, as the cosign verifier.
+
+## Concepts
+
+Artifact signing: To sign an artifact, a signer generates a private/public key pair and uses the secret key to sign an arbitrary piece of data.
+
+Artifact Log: Record of artifact metadata created by signers.
+
+Identity Log: Record of mappings from identities to signing keys.
+
+OpenID Connect (OIDC): A widely-supported protocol allowing relying parties (applications) to verify the identity of resource owners (end users) as confirmed by identity providers.
+
+Identity Provider: Mechanism vouching that an entity (individual) controls an identity (e.g., email account)
+
+Signers: Individuals vouching for the authenticity of content.
+
+Verifiers: Individuals checking that content is authentic.
+
+Certificate Authority(CA): Entity verifying identity and issues cryptographic certificates to signers.
+
+Webpublickeyinfrastructure (PKI): A mature and widely deployed trust ecosystem. At its core, web PKI uses X.509 certificates to pin the trust of web servers to a “certificate authority” (or CA).
+
+## Scenarios
+
+`sigstore-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources. These scenarios can be categorized into the following main types.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Fulcio issues short-lived certificates binding an ephemeral key to an OpenID Connect identity. Signing events are logged in Rekor, a signature transparency log. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Environments where signatures are verified with a known public key. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. Not standalone verification. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. Not standalone verification. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+
+### Verification Input
```

Comment:
input/output could be different from different perspective. As a Ratify developer, I might be interested in the input/output between Ratify and the cosign verifier. As a Ratify user, I might wanna know the input/output to Ratify.

Reply:
Offline discussed.